app/agents_v2/primary_agent/llm_factory.py

```
@@ -0,0 +1,189 @@
+"""
+MB-Sparrow - Primary Agent LLM Factory
+
+This module provides factory methods for instantiating LLM models
+based on the selected model type, supporting Gemini and OpenRouter models.
+"""
+
+import os
+import logging
+from typing import Optional, Dict, Any
```

Ruff (0.12.2)

`typing.Optional` imported but unused

Remove unused import: `typing.Optional`

(F401)